pkg/tcpip/network/ipv6/icmp.go

```diff
@@ -355,6 +355,18 @@ func (e *endpoint) handleICMP(r *stack.Route, netHeader buffer.View, pkt tcpip.P
 
 	case header.ICMPv6RouterSolicit:
 		received.RouterSolicit.Increment()
+		rs := header.NDPRouterSolicit(h.NDPPayload())
```

Comment:
Lets also make sure that `h.NDPPayload()` is at least `header.NDPRSMinimumSize` bytes before using/creating the `header.NDPRouterSolicit`. Also please include a test for this scenario